src/edu/cs505/g5/SubjectDHTSensor.java

```diff
@@ -0,0 +1,46 @@
+p
+package edu.ccsu.cs505.g5;
+/**
+ *
+ * @author CS505-Group5
+ */
+import java.util.Vector;
+/**
+ * This class is the Subject class in the observer pattern which will notify the observers if the temperature goes beyond the limit-set
+ */
+public class SubjectDHTSensor {

```

Comment:
you should have an abstract base class for your sensor which just hold the observers, adding/removing them and the notification method

Reply:
modified